src/tests/Interop/NativeLibrary/API/GetMainProgramHandleTests.cs

```diff
@@ -0,0 +1,127 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Xunit;
+using static TestHelpers;
+
+class GetMainProgramHandleTests
```

Comment:
I'd like to see a test that validates passing the result from `GetMainProgramHandle()` to `Free()`. We should then also document the behavior unless it is a nop - which is what I'd assume it to be.